WordPress/Classes/Services/EditorSettingsService.swift

```diff
@@ -0,0 +1,51 @@
+import Foundation
+
+@objc enum EditorSettingsServiceError: Int, Swift.Error {
+    case mobileEditorNotSet
+}
+
+@objc class EditorSettingsService: LocalCoreDataService {
+    @objc func syncEditorSettings(for blog: Blog, success: @escaping () -> Void, failure: @escaping (Swift.Error) -> Void) {
+        guard let api = blog.wordPressComRestApi(), let siteID = blog.dotComID?.intValue else {
+            let error = NSError(domain: "EditorSettingsService", code: 0, userInfo: [NSDebugDescriptionErrorKey: "Api or dotCom Site ID not found"])
+            failure(error)
+            return
+        }
+
+        let service = EditorServiceRemote(wordPressComRestApi: api)
+        service.getEditorSettings(siteID, success: { (settings) in
+            do {
+                try self.saveRemoteEditorSettings(settings, on: blog)
+            } catch {
+                failure(error)
```

Comment:
While reviewing #12166, I found that this will cause a crash. When the mobile setting is not set, the `saveRemoteEditorSettings` will throw an error. This will call the `failure` handler but it doesn't return, so it will also call `success` a couple of lines later.

This means that the handler in `BlogService` will call `dispatch_group_leave` once too many times and the app will crash.

Reply:
Good catch @koke !
I got that crash while working on the other PRs, and it was fixed here: https://github.com/wordpress-mobile/WordPress-iOS/pull/12189

Good thing that we are working on a feature branch :)